packages/procedures/command_handler/post/index.js

```diff
@@ -85,6 +85,8 @@ module.exports = ({
       await fn();
     }
 
+    if (thenFn) await thenFn();
```

Comment:
await the thenFn. Let the fn's definer decide what should be async and what should be sync. This lets the definer make sure all functionality is at least in flight before returning from this POST.

Reply:
This looks like a reasonable solution to me